src/Infrastructure/Processor/Process/MediaTranslationShopware6ExportProcess.php

```diff
@@ -0,0 +1,150 @@
+use Ergonode\ExporterShopware6\Domain\Entity\Shopware6Channel;
+use Ergonode\ExporterShopware6\Domain\Repository\LanguageRepositoryInterface;
+use Ergonode\ExporterShopware6\Domain\Repository\ProductRepositoryInterface;
+use Ergonode\ExporterShopware6\Infrastructure\Connector\Action\Media\GetMediaTranslations;
+use Ergonode\ExporterShopware6\Infrastructure\Connector\Action\Media\PatchMediaAction;
+use Ergonode\ExporterShopware6\Infrastructure\Connector\Shopware6Connector;
+use Ergonode\ExporterShopware6\Infrastructure\Connector\Shopware6QueryBuilder;
+use Ergonode\ExporterShopware6\Infrastructure\Model\Media\Shopware6MediaTranslation;
+use Ergonode\ExporterShopware6\Infrastructure\Model\Shopware6Media;
+use Ergonode\Multimedia\Domain\Entity\AbstractMultimedia;
+use GuzzleHttp\Exception\ClientException;
+use GuzzleHttp\Exception\GuzzleException;
+use Psr\Log\LoggerInterface;
+use Throwable;
+use Webmozart\Assert\Assert;
+
+class MediaTranslationShopware6ExportProcess
+{
+    private Shopware6Connector $connector;
+
+    protected ProductRepositoryInterface $shopwareProductRepository;
+
+    private LanguageRepositoryInterface $languageRepository;
+
+    private ExportRepositoryInterface $exportRepository;
+
+    private LoggerInterface $logger;
+
+    public function __construct(
+        Shopware6Connector $connector,
+        LanguageRepositoryInterface $languageRepository,
+        ExportRepositoryInterface $exportRepository,
+        ProductRepositoryInterface $shopwareProductRepository,
+        LoggerInterface $logger
+    ) {
+        $this->connector = $connector;
+        $this->languageRepository = $languageRepository;
+        $this->exportRepository = $exportRepository;
+        $this->shopwareProductRepository = $shopwareProductRepository;
+        $this->logger = $logger;
+    }
+
+    public function process(
+        Export $export,
+        Shopware6Channel $channel,
+        AbstractMultimedia $multimedia,
+        string $shopwareId
+    ): void {
+        try {
+            $this->ensureMediaTranslations($channel, $multimedia, $shopwareId);
+        } catch (Throwable $exception) {
+            $this->exportRepository->addError($export->getId(), $exception->getMessage(), []);
+        }
+    }
+
+    /**
+     * @throws GuzzleException
+     */
+    private function ensureMediaTranslations(
+        Shopware6Channel $channel,
+        AbstractMultimedia $multimedia,
+        string $shopwareId
+    ): void {
+        //apply translations from shopware
+        $shopware6Media = new Shopware6Media($shopwareId, null);
+        $shopware6Media->setTranslations($this->getMediaTranslations($channel, $shopwareId));
+        //apply translations from ergo
+        $altTranslations = $multimedia->getAlt()->getTranslations();
+        $titleTranslations = $multimedia->getTitle()->getTranslations();
+        $languages = array_unique(array_merge(array_keys($altTranslations), array_keys($titleTranslations)));
```

Comment:
This logic looks like a potential risk. Are we should all those array operations always return correct data? I would much rather loop through existing translations from Shopware, check if data has changed in Ergonode and apply changes

Reply:
In here we are "converting" / "collecting" ergonode data to later apply on Media Model.
The check that you are mention of is in the Model itself - based on this we are able to get isModified flag